Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/AST/AST.h

```diff
@@ -20,13 +20,31 @@ RefPtr<T> as(NullableTree const& tree)
     return dynamic_cast<T*>(tree.ptr());
 }
 
+class NodeSubtreePointer {
+public:
+    NodeSubtreePointer(Tree* tree_ptr)
+        : m_tree_ptr(tree_ptr)
+    {
+    }
+
+    Tree& get() { return *m_tree_ptr; }
+
+    void replace_subtree(Tree tree) { *m_tree_ptr = move(tree); }
+
+private:
+    Tree* m_tree_ptr;
```

Comment:
I don't really understand this code yet, but this raw pointer is kind of spooky. That being said the code overall looks pretty interesting. Still waiting on test cases :P

Reply:
This subtree pointer is not supposed to be used outside of `RecursiveASTVisitor`, and we control lifetimes pretty well there. Will add `Badge<RecursiveASTVisitor>` here in the future.

And will think what is the best way to test the compiler. Rn all the testing this thing gets is me constantly executing it on the file I attached in the previous PR ;)